helm/cluster/templates/clusterapi/_helpers_flatcar.tpl

```diff
@@ -21,6 +21,23 @@
     {{- if hasKey .contents.unit "defaultDependencies" }}
     DefaultDependencies={{ if .contents.unit.defaultDependencies }}yes{{ else }}no{{ end }}
     {{- end }}
+    {{- if hasKey .contents.unit "after" }}
```

Comment:
I think `hasKey` has been used for the boolean value only. Here you should be fine with...

```suggestion
    {{- if hasKey .contents.unit.after }}
```

Also: wasn't it possible to have more than one `After` in a unit file? I'm not sure... so maybe it would be worth having an array here.

The same of course applies to the other fields.

Reply:
It shouldn't make a difference. `haskey` works on any type and will return true if the object has anything with that key. Though thinking about it we likely want to avoid empty strings so I'll switch it out.

Reply:
Updated

Reply:
Yeah, my concern was about empty strings. Sorry for the confusion!